src/main/kotlin/org/rust/lang/core/resolve/NameResolution.kt

```diff
@@ -779,23 +760,26 @@ private fun processNestedScopesUpwards(scopeStart: RsCompositeElement, processor
     return false
 }
 
-
 // There's already similar functions in TreeUtils, should use it
 private fun walkUp(
-    start: RsCompositeElement,
+    start: PsiElement,
     stopAfter: (RsCompositeElement) -> Boolean,
-    processor: (cameFrom: RsCompositeElement, scope: RsCompositeElement) -> Boolean
+    stopAtFileBoundary: Boolean = false,
+    processor: (cameFrom: PsiElement, scope: RsCompositeElement) -> Boolean
 ): Boolean {
-
-    var cameFrom: RsCompositeElement = start
+    var cameFrom = start
     var scope = start.context as RsCompositeElement?
     while (scope != null) {
         if (processor(cameFrom, scope)) return true
         if (stopAfter(scope)) break
         cameFrom = scope
-        scope = scope.context as RsCompositeElement?
+        scope = scope.context as? RsCompositeElement
+        if (stopAtFileBoundary) {
```

Comment:
Looks like the meaning of this variable is inverted? That is, it says "stop at file boundary", but, when `true`, it actually causes the search to continue beyond the file boundary.

Reply:
Ups